From 842c11cdc0d34bf99bcef95b90837f8ad07b0f5e Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Tue, 8 Nov 2016 13:18:50 -0600
Subject: [PATCH 1309/5242] MXSCM-220 dts: imx: add missing properties for
 wifi on imx6 scm boards

commit  5dad8cb9fe1ed531a952fd43510652cecec6b223 from
https://source.codeaurora.org/external/imx/linux-imx.git

BCM WiFi driver needs to take care of card detect by itself. Using cd-post
property tells the mmc core to not detect the card automatically during
host driver probe and post it untill the client driver notifies to do it.

The non-removable option is also required to fix a NULL dereference
occurred when resuming from suspend. The pm-ignore-notify parameter is
also included.

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dqscm-qwks-wifi.dtsi  |    3 +++
 arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi |    3 +++
 2 files changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/imx6dqscm-qwks-wifi.dtsi b/arch/arm/boot/dts/imx6dqscm-qwks-wifi.dtsi
index 4b29427..bafca74 100644
--- a/arch/arm/boot/dts/imx6dqscm-qwks-wifi.dtsi
+++ b/arch/arm/boot/dts/imx6dqscm-qwks-wifi.dtsi
@@ -87,5 +87,8 @@
 	pinctrl-0 = <&pinctrl_wifi>;
 	bus-width = <4>;
 	no-1-8-v;
+	non-removable;
+	cd-post;
+	pm-ignore-notify;
 	wifi-host;
 };
diff --git a/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi b/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
index 479983e..f55ac0d 100644
--- a/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
+++ b/arch/arm/boot/dts/imx6sxscm-evb-btwifi.dtsi
@@ -106,6 +106,9 @@
 &usdhc2 {
 	pinctrl-0 = <&pinctrl_wifi>;
 	bus-width = <4>;
+	non-removable;
+	cd-post;
+	pm-ignore-notify;
 };
 
 &usdhc3 {
-- 
1.7.9.5

